Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocListImplementation/Rendering/BocColumnRendererArrayBuilderTest.cs

```diff
@@ -57,7 +57,32 @@ public void GetColumnRenderers ()
     }
 
     [Test]
-    public void GetColumnRenderers_WithBocValidationErrorIndicatorColumnDefinition_AlwaysVisibility_WithoutFailures_ContainsNullColumnRenderer ()
+    public void CreateColumnRenderers_WithNotVisibleColumn_CreatesNullColumnRenderer ()
+    {
+      var notVisibleColumn = new StubColumnDefinition { IsVisible = false };
+      var firstColumn = new StubColumnDefinition();
+      var thirdColumn = new StubColumnDefinition();
+
+      var builder = new BocColumnRendererArrayBuilder(CreateColumnCollection(firstColumn, notVisibleColumn, thirdColumn), _serviceLocator);
+
+      var bocColumnRenderers = builder.CreateColumnRenderers();
+
+      Assert.That(bocColumnRenderers.Length, Is.EqualTo(3));
+      Assert.That(PrivateInvoke.GetNonPublicField(bocColumnRenderers[1], "_columnRenderer"), Is.TypeOf(typeof(NullColumnRenderer)));
+      Assert.That(bocColumnRenderers[1].ColumnDefinition, Is.SameAs(notVisibleColumn));
```

Comment:
Question: you want to order the assertions to match the index in the collection or keep it this way? 

I usally order based on the index to make it easier to reason over the order of items. The currently implemented variant emphazies the most important aspect of the test but makes it more difficult to see the correct order of the index. It's a matter of style, mostly. Unless you think: yes, true, should do it the other way around.

Reply:
It was a concious decision to to it like this, I think it makes most sense here